doc/build/kconfig/setting.rst

```diff
@@ -15,6 +15,37 @@ All Kconfig options can be searched in the :ref:`Kconfig search page
    Before making changes to Kconfig files, it's a good idea to also go through
    the :ref:`kconfig_tips_and_tricks` page.
 
+.. _configuration_file_types:
+
+Kconfig configuration file types
+********************************
+
+There are two types of Kconfig configuration files:
```

Comment:
I think it's very good to have the concept and highlevel description of Kconfig and config files introduced early.

But instead of repeating what is described later, then perhaps change the title of:
https://github.com/zephyrproject-rtos/zephyr/blob/18b18713515cdf2ad5783a19b1b9905690be5b53/doc/build/kconfig/setting.rst?plain=1#L92-L93 to something else, like `Configuration files`, and then link to that section.

Similar with the `Kconfig files`, create a generic subsection of how the language looks and link to it, cause afterwards the use of the language is presented in the `Visible and invisible Kconfig symbols` section.

The introduction to Kconfig language itself in a generic way can then also link to the Kconfig language for further details.
See also the discussion in this old proposal: https://github.com/zephyrproject-rtos/zephyr/pull/38746

Reply:
This can give an overview like:
```
Kconfig configuration file types
********************************

There are two types of Kconfig configuration files:
1. **Kconfig Files**: which describes the Kconfig tree by defining symbols. Details of Kconfig files are found in <link>
2. **.config Files**: which configures the Kconfig tree by assigning values to symbols. Details of configuration files are found in <link>.
```